### PR TITLE
User and Credits

### PR DESCRIPTION
**Pull request by theo-damiani**

### CODE REVIEW

File: app/src/main/java/com/github/steroidteam/todolist/viewmodel/UserViewModel.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    public LiveData<FirebaseUser> getUser() {
+        if (userLiveData == null) {
+            userLiveData = new MutableLiveData<>();
+            loadUser();
+        }
+        return userLiveData;
+    }
+
+    private void loadUser() {
+        System.err.println("ON LOAD !!!!!!!!!!!!!!!!! <<<<<<<<<<<<<<<<<<<<<<<<<<<<<");
```
Comment (YagoGG) [inline app/src/main/java/com/github/steroidteam/todolist/viewmodel/UserViewModel.java:35]: Debug statement, should be removed.

File: app/src/main/java/com/github/steroidteam/todolist/viewmodel/UserViewModel.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public void updateUser(UserProfileChangeRequest request) {
+        if (userLiveData.getValue() != null) {
+            userLiveData
+                    .getValue()
+                    .updateProfile(request)
+                    .addOnSuccessListener(
+                            v -> {
+                                System.out.println("=================================>");
```
Comment (YagoGG) [inline app/src/main/java/com/github/steroidteam/todolist/viewmodel/UserViewModel.java:46]: Debug statement, should be removed.

File: app/src/main/res/layout/fragment_profile.xml [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/profile_mail_edit">
+
+            <Button
+                android:id="@+id/profile_pwd_edit_btn"
+                android:layout_width="match_parent"
+                android:layout_height="0dp"
+                android:layout_weight=".3"
+                style="@style/TaskSaveButton"
+                android:text="Modify password:" />
```
Comment (YagoGG) [inline app/src/main/res/layout/fragment_profile.xml:251]: Unwanted colon (`:`) at the end.

File: app/src/main/java/com/github/steroidteam/todolist/view/dialog/AuthDialogFragment.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import android.view.View;
+import android.widget.EditText;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.view.ContextThemeWrapper;
+import androidx.fragment.app.DialogFragment;
+import com.github.steroidteam.todolist.R;
+
+public class AuthDialogFragment extends DialogFragment {
+
+    private final String DIALOG_TITLE = "Please, you need to re-authenticate!";
```
Comment (YagoGG) [inline app/src/main/java/com/github/steroidteam/todolist/view/dialog/AuthDialogFragment.java:17]: Move this to `strings.xml` and then do `R.strings.X`.

**Review by YagoGG — CHANGES_REQUESTED** [review]
Hey @theo-damiani! Thanks for the changes, I left a few minor inline comments.

I am a bit confused as to why the user has to re-authenticate. Is it an intentional security feature? For what I saw in the code, it does not seem to be required by Firebase (but I might have misunderstood it).

**Review by YagoGG — APPROVED** [review]
Made the changes with Théo, it looks great(er) now. Cheers!

**Review by VPellenc — APPROVED** [review]
It works very well, thanks for the feature!

**Pull request merged**